vars/cloudFoundryDeploy.groovy

```diff
@@ -222,6 +222,24 @@ void call(Map parameters = [:]) {
         //make sure that for further execution whole workspace, e.g. also downloaded artifacts are considered
         config.stashContent = []
 
+        // validate cf app name
+        if (config.cloudFoundry.appName) {
+            String appName = config.cloudFoundry.appName.toString()
+            boolean isValidCfAppName = appName.matches("^[a-zA-Z0-9]*\$")
```

Comment:
I would like to suggest to allow also the dash (`-`) in the app name, since there is an example also using a dash in the link provided below. I _think_ the dash is the only allowed character beside what is already contained in the regex. And it must not start or end with a dash ...

>For example, an app named my-app-1234 running on CF with an apps domain apps.example.com would, by default, run at the URL https://my-app-1234.apps.example.com.